packages/api-fetch/src/test/index.js

```diff
@@ -4,53 +4,53 @@
 import apiFetch from '../';
 
 /**
- * Mock return value for a successful fetch JSON return value.
+ * Mock response value for a successful fetch.
  *
- * @return {Promise} Mock return value.
+ * @return {Response} Mock return value.
  */
-const DEFAULT_FETCH_MOCK_RETURN = Promise.resolve( {
+const DEFAULT_FETCH_MOCK_RETURN = {
+	ok: true,
 	status: 200,
 	json: () => Promise.resolve( {} ),
-} );
+};
 
 describe( 'apiFetch', () => {
-	const originalFetch = window.fetch;
+	const originalFetch = globalThis.fetch;
 
 	beforeEach( () => {
-		window.fetch = jest.fn();
+		globalThis.fetch = jest.fn();
 	} );
 
 	afterAll( () => {
-		window.fetch = originalFetch;
+		globalThis.fetch = originalFetch;
 	} );
 
-	it( 'should call the API properly', () => {
-		window.fetch.mockReturnValue(
-			Promise.resolve( {
-				status: 200,
-				json() {
-					return Promise.resolve( { message: 'ok' } );
-				},
-			} )
-		);
+	it( 'should call the API properly', async () => {
+		globalThis.fetch.mockResolvedValue( {
+			ok: true,
+			status: 200,
+			async json() {
+				return { message: 'ok' };
+			},
+		} );
 
-		return apiFetch( { path: '/random' } ).then( ( body ) => {
-			expect( body ).toEqual( { message: 'ok' } );
+		await expect( apiFetch( { path: '/random' } ) ).resolves.toEqual( {
+			message: 'ok',
 		} );
 	} );
 
-	it( 'should fetch with non-JSON body', () => {
-		window.fetch.mockReturnValue( DEFAULT_FETCH_MOCK_RETURN );
+	it( 'should fetch with non-JSON body', async () => {
+		globalThis.fetch.mockResolvedValue( DEFAULT_FETCH_MOCK_RETURN );
 
 		const body = 'FormData';
 
-		apiFetch( {
+		await apiFetch( {
```

Comment:
Example of a synchronous test where the response promise was leaking out of the test. The promise resolves only after the test has already finished running.